Library/Homebrew/cmd/install-vendor.sh

```diff
@@ -0,0 +1,108 @@
+source "$HOMEBREW_LIBRARY/Homebrew/utils/lock.sh"
+
+if [[ -n "$HOMEBREW_OSX" ]]
+then
+  Git_URL=""
+  Git_SHA=""
+  Ruby_URL=""
+  Ruby_SHA=""
+else
+  Git_URL=""
+  Git_SHA=""
+  Ruby_URL=""
+  Ruby_SHA=""
+fi
+
+cleanup-old() {
+  for version in "$HOMEBREW_LIBRARY/Vendor/$VENDOR_NAME/Versions"/*
+  do
+    [[ -L "$version" ]] && continue # skip Current symlink
+    rm -rf "$version"
+  done
+}
+
+install() {
+  local curl_args
+  local tar_args
+  local sha
+
+  VENDOR_FILE="$(/usr/bin/mktemp "/tmp/homebrew-$VENDOR_NAME.XXXXXX")"
+  trap '{ rm -f "$VENDOR_FILE"; unlock "install-vendor-$VENDOR_NAME"; exit 1; }' EXIT
+
+  if [[ -n "$HOMEBREW_VERBOSE" ]]
+  then
+    curl_args="-fLA"
+    tar_args="xvzf"
+  elif [[ -n "$HOMEBREW_QUIET" ]]
+  then
+    curl_args="-sfLA"
+    tar_args="xzf"
+  else
+    curl_args="-#fLA"
+    tar_args="xzf"
+  fi
+  [[ -z "$HOMEBREW_QUIET" ]] && echo "==> Downloading $VENDOR_URL"
+  curl "$curl_args" "$HOMEBREW_USER_AGENT_CURL" -o "$VENDOR_FILE" "$VENDOR_URL"
+
+  if [[ -x "/usr/bin/shasum" ]]
+  then
+    sha="$(/usr/bin/shasum -a 256 "$VENDOR_FILE" | cut -d' ' -f1)"
+  elif [[ -n "$(which sha256sum)" ]]
+  then
+    sha="$(sha256sum "$VENDOR_FILE" | cut -d' ' -f1)"
+  else
+    odie "Cannot verify the checksum."
+  fi
+
+  if [[ "$sha" != "$VENDOR_SHA" ]]
+  then
+    odie "Checksum mismatch."
+  fi
+
+  chdir "$HOMEBREW_LIBRARY/Vendor/$VENDOR_NAME/Versions"
+  [[ -z "$HOMEBREW_QUIET" ]] && echo "==> Install $(basename "$VENDOR_URL")"
+  tar "$tar_args" "$VENDOR_FILE"
+  rm -f "$VENDOR_FILE"
+  trap - EXIT
+}
+
+homebrew-install-vendor() {
+  local option
+  local url_var
+  local sha_var
+
+  for option in "$@"
+  do
+    case "$option" in
+      -\?|-h|--help|--usage) brew help install-vendor; exit $? ;;
+      --verbose) HOMEBREW_VERBOSE=1 ;;
+      --quiet) HOMEBREW_QUIET=1 ;;
+      --debug) HOMEBREW_DEBUG=1 ;;
+      --*) ;;
+      -*)
+        [[ "$option" = *v* ]] && HOMEBREW_VERBOSE=1;
+        [[ "$option" = *q* ]] && HOMEBREW_QUIET=1;
+        [[ "$option" = *d* ]] && HOMEBREW_DEBUG=1;
+        ;;
+      *)
+        [[ -n "$VENDOR_NAME" ]] && odie "This command does not take multiple vendor targets"
+        VENDOR_NAME="$(tr '[:lower:]' '[:upper:]' <<< "${option:0:1}")${option:1}"
+        ;;
+    esac
+  done
+
+  [[ -z "$VENDOR_NAME" ]] && odie "This command requires one vendor target."
+  [[ -n "$HOMEBREW_DEBUG" ]] && set -x
+
+  url_var="${VENDOR_NAME}_URL"
+  sha_var="${VENDOR_NAME}_SHA"
+  VENDOR_URL="${!url_var}"
+  VENDOR_SHA="${!sha_var}"
+
+  [[ -z "$VENDOR_URL" || -z "$VENDOR_SHA" ]] && odie "Unknown vendor target"
+
+  lock "install-vendor-$VENDOR_NAME"
+  cleanup-old
```

Comment:
> Have a file e.g. vendor/ruby-version that contains the latest version rather than the opt symlink and only move the symlink or remove the last version when that version is known to be working.

My current question is how to handle it for git. For example, if we add following code to `ENV/scm/git`
```ruby
if File.exist?(HOMEBREW_LIBRARY/"vendor/git/opt/bin/git")
  exec HOMEBREW_LIBRARY/"vendor/git/opt/bin/git", *ARGV
end
```
But the `ENV/scm/git` should not invoke `brew install-vendor git`, it should be called by `brew update` or `Utils.ensure_git_installed!`.